### PR TITLE
[1.21] Kill `GameData`

### LINKED ISSUE #1078: Kill `GameData`
Shouldn't we move its methods to `RegistryManager`?

### PR DESCRIPTION
**Pull request by Spinoscythe**
Fixes #1078

### CODE REVIEW

File: patches/net/minecraft/world/level/block/Block.java.patch [hunk 1]
```diff
@@ -11,7 +11,7 @@
      private final Holder.Reference<Block> builtInRegistryHolder = BuiltInRegistries.BLOCK.createIntrusiveHolder(this);
 -    public static final IdMapper<BlockState> BLOCK_STATE_REGISTRY = new IdMapper<>();
 +    @Deprecated //Forge: Do not use, use GameRegistry
```
Comment (PepperCode1) [inline patches/net/minecraft/world/level/block/Block.java.patch]: Should this still be deprecated?

Comment (Spinoscythe) [reply patches/net/minecraft/world/level/block/Block.java.patch]: Im not too sure about that unfortunately

File: patches/net/minecraft/world/level/block/Block.java.patch [hunk 2]
```diff
@@ -11,7 +11,7 @@
...
-+    public static final IdMapper<BlockState> BLOCK_STATE_REGISTRY = net.neoforged.neoforge.registries.GameData.getBlockStateIDMap();
++    public static final IdMapper<BlockState> BLOCK_STATE_REGISTRY = net.neoforged.neoforge.registries.RegistryManager.getBlockStateIdMap();
```
Comment (Soaryn) [inline patches/net/minecraft/world/level/block/Block.java.patch:13]: Just checking to make sure this was the replacement to `GameData.getBlockStateIDMap()`, while it is pretty niche, I do need a full list of blockstates on load to handle an odd recipe presenter in something like JEI. It LOOKS to be the replacement, just wanted to be sure.

(sleep deprived cursory glance tells me it is just moving to RegistryManager, and this is an accessor, so was perhaps a bit confused as to why have both. No qualms with them both existing, just wanted clarifications :) )

Comment (Spinoscythe) [reply patches/net/minecraft/world/level/block/Block.java.patch:13]: It is indeed just moving to RegistryManager

**Comment by TelepathicGrunt** [comment]
Looks like this picked up a merge conflict somewhere along the way. The conflict will need to be fixed first

**Comment by Spinoscythe** [comment]
Messed up this PR. Will make a new one

**Pull request closed without merging**